src/cmd/tools/read_data_files/main/main.go

```diff
@@ -180,7 +186,17 @@ func main() {
 				data = entry.Data
 			)
 
-			if *idFilter != "" && !strings.Contains(id.String(), *idFilter) {
+			var filterMatches bool = false
+			var matchedId string = ""
+			for _, filterId := range filterIds {
```

Comment:
could we optimize this for loop with a pre-compiled regex? and then do a regex match? so in the future the filter can just be `metric1|metric2|metric3` etc?

Reply:
anyway, this might not be that important.

Reply:
fixed